### PR TITLE
Move usb-remap internals for certain ICs to USB enable()

### PR DESCRIPTION
**Pull request by brainstorm**
Hopefully fixes issue #118 ?

### CODE REVIEW

File: src/usb.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ unsafe impl UsbPeripheral for Peripheral {
 
     fn enable() {
         let rcc = unsafe { (&*RCC::ptr()) };
```
Comment (brainstorm) [inline src/usb.rs:30]: Newbie question, @therealprof: can this reference to the pointer of RCC `unsafe` be avoided somehow? I'm curious.

**Comment by mvirkkunen** [comment]
I think it would be preferable to avoid extra feature flags. Would it be possible to implement this using a separate `Peripheral` type and/or constructor?

File: src/usb.rs (hunk 1 above)
Comment (therealprof) [reply src/usb.rs:30]: Only if you require passing in `RCC` into your function or putting the method on the "owner" of RCC.

**Comment by therealprof** [comment]
I concur with @mvirkkunen and would prefer this to be not dependent on a feature flag.

However the elephant in the room is certainly the enabling of the `SYSCFG` peripheral. It is a quite important peripheral but then again maybe not that important that we would want to unconditionally enable it? Hm...

**Comment by brainstorm** [comment]
Totally broken code passes? hm...

**Comment by brainstorm** [comment]
No worries @therealprof, this got me acquainted with Rust embedded HALs for the first time, so I shall thank you ;)

Tusen tack!

**Comment by therealprof** [comment]
Thanks for sticking around!

**Comment by brainstorm** [comment]
My pleasure! Would you mind pushing a 0.17.1 release for this? :)

**Comment by therealprof** [comment]
Can do.

**Comment by mvirkkunen** [comment]
N.b. the commented out code in the example is missing the arguments

**Pull request merged**